scenario/smartsearch.go

```diff
@@ -125,6 +208,115 @@ func parseSearchTerms(s string) []string {
 	return searchTerms
 }
 
+func doSmartSearchRPCs(sessionState *session.State, actionState *action.State, appDoc *enigma.Doc, id int, searchText string) {
+	searchTerms := parseSearchTerms(searchText)
+	if sessionState.LogEntry.ShouldLogDebug() {
+		sessionState.LogEntry.LogDebugf(`search%d text %s becomes search terms: %s`, id, quote(searchText), quoteList(searchTerms))
+	}
+
+	sessionState.QueueRequest(func(ctx context.Context) error {
+		searchSuggestionResult, err := appDoc.SearchSuggest(
+			ctx,
+			&enigma.SearchCombinationOptions{},
+			searchTerms,
+		)
+		logSearchSuggestionResult(sessionState.LogEntry, id, searchSuggestionResult)
+		return err
+	}, actionState, true, "SearchSuggest call failed")
+
+	sessionState.QueueRequest(func(ctx context.Context) error {
+		searchResult, err := appDoc.SearchResults(
+			ctx,
+			&enigma.SearchCombinationOptions{
+				Context:      "CurrentSelections",
+				CharEncoding: "Utf16",
+			},
+			searchTerms,
+			&enigma.SearchPage{
+				Offset: 0,
+				Count:  5,
+				GroupOptions: []*enigma.SearchGroupOptions{
+					{
+						GroupType: "DatasetType",
+						Offset:    0,
+						Count:     intPtr(-1),
+					},
+				},
+				GroupItemOptions: []*enigma.SearchGroupItemOptions{{
+					GroupItemType: "Field",
+					Offset:        0,
+					Count:         intPtr(5),
+				}},
+			},
+		)
+		logSearchResult(sessionState.LogEntry, id, searchResult)
+		return err
+	}, actionState, true, "SearchResults call failed")
+}
+
+func newSearchTextChunks(randomizer helpers.Randomizer, searchText string, pasteSearchText bool) (searchTextChunks, error) {
+	if pasteSearchText {
+		return searchTextChunks{
+			searchTextChunk{
+				Text:                    searchText,
+				PostTypingThinkDuration: 0,
+				TypingDuration:          0,
+			},
+		}, nil
+	}
+
+	const typeOneCharDuration = 300 * time.Millisecond
```

Comment:
It's very slow typing

Reply:
also in general maybe some comments to the chosen times.

Reply:
Alright, will add comments. This is based on typing speed 40 words per minute and average english word length 4.7.
60 * 1000 / (40 * 4.7) ~= 300 ms

But with the other breaks i agree it is a bit slow.